src/test.rs

```diff
@@ -29,13 +29,23 @@
 //! }
 //! ```
 //!
-//! We can test some requests against the `sum` filter like this:
+//! We can test some requests against the `sum` filter like this (by tag your async test function with `#[tokio::test]`):
 //!
 //! ```
 //! # use warp::Filter;
-//! #[tokio::test]
+//! # #[tokio::main]
+//! # async fn main() {
+//! #   test_sum().await;
+//! # }
+//! #
+//! # fn sum() -> impl Filter<Extract = (u32,), Error = warp::Rejection> + Copy {
+//! #     warp::path::param()
+//! #         .and(warp::path::param())
+//! #         .map(|x: u32, y: u32| {
+//! #             x + y
+//! #         })
+//! # }
 //! async fn test_sum() {
-//! #    let sum = || warp::any().map(|| 3);
```

Comment:
Firstly, previously we use `#[tokio::test]` and that is actually not really working with `cargo test ---doc`, which basically hides your test and nothing is running.

I had to switch `#[tokio::test]` to `#[tokio::main]` to make the test actually running. Also I think it's nicer to test a real `sum()` filter instead of this fake one here, so I created the `sum` filter above but hidden from the docs. So this still make the separate blocks compile and also make it actually running now while previous it didn't

Reply:
> Firstly, previously we use #[tokio::test] and that is actually not really working with cargo test ---doc, which basically hides your test and nothing is running.

currently it's `#[test]`, which yeah should be updated to `#[tokio::test]` along the `await` on the request calls. 

> I had to switch #[tokio::test] to #[tokio::main] to make the test actually running. Also I think it's nicer to test a real sum() filter instead of this fake one here, so I created the more realistic sum filter above but hidden from the docs. So this still make the separate blocks compile and also make it actually running now while previous it didn't.

Yes, but the test module doc examples should show how to write a test case, eg: with `#[tokio::test]` above the function definition

Reply:
Thing with using `#[tokio::test]` will make the doc test basically invisible and not ran during `cargo test test --doc`. Is that something we want or?

Reply:
yeah, that doc serve as examples on how users can write tests for warp projects